src/CsvWriter.php

```diff
@@ -0,0 +1,150 @@
+        $this->openCsvFile();
+    }
+
+    public function __destruct()
+    {
+        $this->closeFile();
+    }
+
+    /**
+     * @param array $row
+     * @throws Exception
+     */
+    public function writeRow(array $row)
+    {
+        $str = $this->rowToStr($row);
+        $ret = fwrite($this->getFilePointer(), $str);
+
+        /* According to http://php.net/fwrite the fwrite() function
+         should return false on error. However not writing the full
+         string (which may occur e.g. when disk is full) is not considered
+         as an error. Therefore both conditions are necessary. */
+        if (($ret === false) || (($ret === 0) && (strlen($str) > 0))) {
+            throw new Exception(
+                "Cannot write to file {$this->getPathname()}",
+                Exception::WRITE_ERROR,
+                null,
+                Exception::WRITE_ERROR_STR
+            );
+        }
+    }
+
+    /**
+     * @param array $row
+     * @return string
+     * @throws Exception
+     */
+    public function rowToStr(array $row)
+    {
+        $return = [];
+        foreach ($row as $column) {
+            if (!(
+                is_scalar($column)
+                || is_null($column)
+                || (
+                    is_object($column)
+                    && method_exists($column, '__toString')
+                )
+            )) {
+                $type = gettype($column);
+                throw new Exception(
+                    "Cannot write {$type} into a column",
+                    Exception::WRITE_ERROR,
+                    null,
+                    Exception::WRITE_ERROR_STR,
+                    ['column' => $column]
+                );
+            }
+
+            $return[] = $this->getEnclosure() .
+                str_replace($this->getEnclosure(), str_repeat($this->getEnclosure(), 2), $column) .
+                $this->getEnclosure();
+        }
+        return implode($this->getDelimiter(), $return) . "\n";
+    }
+
+    /**
+     * @return resource
+     */
+    protected function getFilePointer()
+    {
+        return $this->filePointer;
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function openCsvFile()
+    {
+        $this->filePointer = @fopen($this->getPathname(), $this->mode);
+        if (!$this->filePointer) {
+            throw new Exception(
+                "Cannot open file {$this->getPathname()} " . error_get_last()['message'],
+                Exception::FILE_NOT_EXISTS,
+                null,
+                Exception::FILE_NOT_EXISTS_STR
+            );
+        }
+    }
+
+    protected function closeFile()
+    {
+        if (is_resource($this->filePointer)) {
+            fclose($this->filePointer);
+        }
+    }
+
+    protected function setMode($mode)
+    {
+        $this->validateMode($mode);
+        $this->mode = $mode;
+    }
+
+    protected function validateMode($mode)
+    {
+        $allowedModes = ['w', 'w+', 'a', 'a+', 'x'];
```

Comment:
bylo by hezké mít na to descriptive konstanty v tom Writeru. Plus mi přijde, že ty s "+" nemají smysl, když je to jen writer. 

OVERWRITE
APPEND
CREATE AND OVERWRITE (?)

Reply:
nevim no, jsou to standardni php stringy, kdyz pro ne nema konstanty PHP, prijde mi trosku divny je delat - evokuje to, to, ze to je "neco jinyho", jakoze fopen(..., 'w') je standardni konstrukce, csvfile->open(..., OVERWRITE) vypada, ze dela neco jinyho

pluska davam pryc